### PR TITLE
Add area filter selector for default dashboard

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add area filter selector for default dashboard. Clicking on the field will open a dialog to hide and re-order area.

CleanShot 2023-11-28 at 18 52 59@2x

CleanShot 2023-11-27 at 15 21 57

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -5266,7 +5271,7 @@
           },
           "strategy": {
             "original-states": {
-              "hidden_areas": "Hidden Areas",
+              "areas": "Displayed areas",
```
Comment (piitaya) [inline src/translations/en.json:5274]:
```suggestion
              "areas": "Areas",
```

File: src/components/ha-area-filter.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+export class HaAreaPicker extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public label?: string;
+
+  @property({ attribute: false }) public value?: AreaFilterValue;
+
+  @property() public helper?: string;
+
+  @property({ type: Boolean }) public disabled?: boolean;
```
Comment (bramkragten) [inline src/components/ha-area-filter.ts:25]:
```suggestion
  @property({ type: Boolean }) public disabled = false;
```

File: src/components/ha-area-filter.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  @property({ type: Boolean }) public required?: boolean;
```
Comment (bramkragten) [inline src/components/ha-area-filter.ts:27]:
```suggestion
  @property({ type: Boolean }) public required = false;
```

File: src/components/ha-area-filter.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      <ha-textfield
+        .label=${this.label}
+        .value=${description}
+        helperPersistent
+        .helper=${this.helper}
+        readOnly
+        icon
+        iconTrailing
+        .disabled=${Boolean(this.disabled)}
+        .required=${Boolean(this.required)}
```
Comment (bramkragten) [inline src/components/ha-area-filter.ts:52]:
```suggestion
        .disabled=${this.disabled}
        .required=${this.required}
```

File: src/dialogs/area-filter/area-filter-dialog.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+      return nothing;
+    }
+
+    const allAreas = this._areas;
+
+    return html`
+      <ha-dialog
+        open
+        @closed=${this._cancel}
+        .heading=${this._dialogParams.title ?? "Areas"}
```
Comment (bramkragten) [inline src/dialogs/area-filter/area-filter-dialog.ts:118]: Should this be lokalized?

File: src/dialogs/area-filter/area-filter-dialog.ts [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+                    class="handle"
+                    .path=${mdiDrag}
+                    slot="graphic"
+                  ></ha-svg-icon>
+                  ${this.hass!.areas[area]?.name || area}
+                  <ha-svg-icon
+                    class="action"
+                    .path=${isVisible ? mdiEye : mdiEyeOff}
+                    slot="meta"
+                  ></ha-svg-icon>
```
Comment (bramkragten) [inline src/dialogs/area-filter/area-filter-dialog.ts:142]: I expected this to be a button, and not the entire row to be clickable?

Comment (piitaya) [reply src/dialogs/area-filter/area-filter-dialog.ts:142]: Yes, we can. Let's discuss that during front-end meeting.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**